### ISSUE TITLE
BERT model is returning NaN logits values in output

### ISSUE THREAD

**Issue by vinayak-shanawad** [triton-inference-server/server#4842]
**Description**
I'm able to deploy fine-tuned "bert-base-uncased" model on Triton inference server using TensorRT, while inference I am getting a NaN logits values.

Converted the onnx model to tensorrt using command below.
trtexec --onnx=model.onnx --saveEngine=model.plan --minShapes=input_ids:1x1,attention_mask:1x1,token_type_ids:1x1 --optShapes=input_ids:16x128,attention_mask:16x128,token_type_ids:16x128 --maxShapes=input_ids:128x128,attention_mask:128x128,token_type_ids:128x128 --fp16 --verbose --workspace=14000 | tee conversion_bs16_dy.txt

**Output logs**
`logits: [[[nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan
   nan nan nan nan]
  ............
  [nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan
   nan nan nan nan]]]`

**Triton Information**
Triton Server Version 2.22.0
NVIDIA Release 22.05

Using Triton container:
'007439368137.dkr.ecr.us-east-2.amazonaws.com/sagemaker-tritonserver:22.05-py3'

**To Reproduce**
1. Deploy tensorrt model on triton inference server.
2. Send a inference request.

`text = "Published by HT Digital Content Services with permission from Construction Digital."`
`batch_size = 1`
`payload = {
    "inputs": [
        {
            "name": "TEXT",
            "shape": (batch_size,),
            "datatype": "BYTES",
            "data": [text],
        }
    ]
}`

Preprocessed the input text, got the input_ids and attention_masks from tokenizer then send the below input to the model.

Model input:

`{'input_ids': array([[  101, 12414, 10151, 87651, 10764, 18491, 12238, 10171, 48822,
        10195, 13154, 10764,   119,   102]], dtype=int32), 'token_type_ids': array([[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=int32), 'attention_mask': array([[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]], dtype=int32)}`

Then you will see that model produce the logits values as NaN.

Please find the all deployment files on G-drive - https://drive.google.com/file/d/1uteEOgnSLwtfTonJtgukKjnDwycezFg3/view?usp=sharing

**Expected behavior**
I expect valid logits values from the BERT model instead of NaN.

Please help me on this issue. Thanks

**Comment by Vinayaks117** [comment]
Hi @alxmamaev @dyastremsky @Tabrizian @rmccorm4

Hope you are doing well. Could anyone please address this issue?

Thanks in advance.

**Comment by krishung5** [comment]
Hi @Vinayaks117, sorry about the late reply. I've been investigating this and I'm trying to reproduce the issue. I'll comment here once I have some update.

**Comment by Vinayaks117** [comment]
No worries @krishung5 Thanks for the confirmation.

**Comment by krishung5** [comment]
@Vinayaks117  Could you run the model using [Polygraphy](https://github.com/NVIDIA/TensorRT/tree/main/tools/Polygraphy) to confirm that the model works outside Triton? This can help us narrow down if the issue occurs inside Triton.

**Comment by Vinayaks117** [comment]
@krishung5 It will take some time for me to run the model on Polygraphy because I didn't use that earlier.

FYI: I was able to deploy onnx model on Triton and able to get the inference with same code which I shared earlier. Thanks

**Comment by rmccorm4** [comment]
> It will take some time for me to run the model on Polygraphy because I didn't use that earlier.

@Vinayaks117 Hopefully something like this should get you started (assuming you're in the same directory as your ONNX model, otherwise change mount paths)

```
# Start latest TRT container that comes with polygraphy installed
docker run -ti --gpus all -v ${PWD}:/mnt -w /mnt nvcr.io/nvidia/tensorrt:22.08-py3

# Let polygraphy install dependencies as needed (onnxruntime, etc)
export POLYGRAPHY_AUTOINSTALL_DEPS=1

# Run model with both onnxruntime, and tensorrt, and then compare the outputs
polygraphy run --onnxrt --trt model.onnx

# For more details, config options, dynamic shape settings, etc.
polygraphy -h
```

**Comment by Vinayaks117** [comment]
Thanks @rmccorm4 for sharing the commands. Please find the results below.

1. onnx model - polygraphy validation is passed.
Results: [onnx-model-validation-results.txt](https://github.com/triton-inference-server/server/files/9540637/onnx-model-validation-results.txt)

2. Converted the onnx model to trt model with minshapes: 1x1 using below command and its polygraphy validation is failed.
`trtexec --onnx=model.onnx --saveEngine=model_bs16.plan --minShapes=input_ids:1x1,attention_mask:1x1,token_type_ids:1x1 --optShapes=input_ids:16x128,attention_mask:16x128,token_type_ids:16x128 --maxShapes=input_ids:128x128,attention_mask:128x128,token_type_ids:128x128 --fp16 --verbose --workspace=14000 | tee conversion_bs16_dy.txt`

Results: [ort-model-minshape-1x1-validation-results.txt](https://github.com/triton-inference-server/server/files/9540642/ort-model-minshape-1x1-validation-results.txt)

3. Converted the onnx model to trt model with minshapes: 1x128 using below command and its polygraphy validation is failed.
`trtexec --onnx=model.onnx --saveEngine=model_bs16.plan --minShapes=input_ids:1x128,attention_mask:1x128,token_type_ids:1x128 --optShapes=input_ids:16x128,attention_mask:16x128,token_type_ids:16x128 --maxShapes=input_ids:128x128,attention_mask:128x128,token_type_ids:128x128 --fp16 --verbose --workspace=14000 | tee conversion_bs16_dy.txt`

Results: [ort-model-minshape-1x128-validation-results.txt](https://github.com/triton-inference-server/server/files/9540646/ort-model-minshape-1x128-validation-results.txt)

Please find the model.onnx file on [G-drive](https://drive.google.com/file/d/1BChlgTqpksj9OpgyLBSv5Qoxj5YZKoGu/view?usp=sharing)

Please let me know if you need any additional details. Thanks

**Comment by rmccorm4** [comment]
Hi @Vinayaks117 ,

For #3 I see validation failed and you get the same all `nan` outputs, which likely means there's an issue with your model, and this is not related to Triton.

You should further investigate your model to verify why it's not working as expected. Polygraphy may be able to help there as well to see which layer in the model the `nan`s first start to appear.

You may want to open an issue with TRT team if you need further help. https://github.com/NVIDIA/TensorRT

**Comment by dyastremsky** [comment]
Closing due to inactivity. Please let us know to reopen the issue if you'd like to follow up.
